### PR TITLE
Initial onboarding UI

### PR DESCRIPTION
**Pull request by stefanhk31**
Functionality is still in progress, but I wanted to give you something that you can work with, since I know we want to get 0.1 out soon. 

Please feel free to add, remove, refactor, etc. to your heart's content, as this is my first foray into Kotlin.

### CODE REVIEW

File: app/src/main/java/org/coepi/android/ui/onboarding/OnboardingViewModel.kt [hunk 1]
```diff
@@ -14,10 +27,33 @@ class OnboardingViewModel(
...
 
+    val context: Context = TODO("Need this to pick up main context, not sure how in VM");
+    val activity: Activity = TODO("Need this to attach to main activity, not sure how in VM");
     val text: LiveData<String> = Observable.just("TODO onboarding").toLiveData()
 
-    fun onCloseClick() {
-        preferences.putBoolean(SEEN_ONBOARDING, true)
+
+    fun onCtaClick() {
+        if (ContextCompat.checkSelfPermission(context, Manifest.permission.BLUETOOTH) != PackageManager.PERMISSION_GRANTED) {
```
Comment (i-schuetz) [inline app/src/main/java/org/coepi/android/ui/onboarding/OnboardingViewModel.kt]: Move this to an own component and wire with the other checks in `BlePreconditions`

File: app/src/main/java/org/coepi/android/ui/onboarding/OnboardingViewModel.kt [hunk 2]
```diff
@@ -14,10 +27,33 @@ class OnboardingViewModel(
     private val preferences: Preferences
 ) : ViewModel() {
 
+    val context: Context = TODO("Need this to pick up main context, not sure how in VM");
+    val activity: Activity = TODO("Need this to attach to main activity, not sure how in VM");
```
Comment (i-schuetz) [inline app/src/main/java/org/coepi/android/ui/onboarding/OnboardingViewModel.kt]: ViewModel shouldn't be coupled with UI. You have to add the permission via `BlePreconditions`

**Comment by stefanhk31** [comment]
Thanks for your suggestions @i-schuetz . I'm afraid I'm still pretty confused, though. I've just pushed my WIP to feature/onboarding, if you want to see where I'm getting stuck. 

Two main questions:

1. I'm still not sure how to inject context into the permissions check, even if I move it into its own component. (Do I even need to check context at all?)
2. What, exactly, do I need to do to wire up to `blePreconditions`? I'm trying to trigger `OnboardingViewModel.closeOnboardingScreen()` once the modal has popped and permission is granted, but I'm sure I'm getting lost in the weeds.

Thanks for your help, and for bearing with me as I dive off the deep end!

**Comment by i-schuetz** [comment]
`BlePreconditions` has `Activity` as a dependency, which is itself a `Context` and also provides the application context. 

`BlePreconditions` also has for the required coarse location permission and enables bluetooth. You have to modify it to request multiple permissions, and wire it such that you can use it from the onboarding. I think that multiple permissions don't work on Android pre-M (please confirm). Somoneone lowered the min API recently, so maybe this has to be reverted.

If becomes too time consuming let me know and I can help.

**Comment by stefanhk31** [comment]
Thanks @i-schuetz . I think I'll definitely need some help with this. Can you push some starter code to get me rolling in the right direction? 

I also may be able to do a quick screen-share, though I'll be working until around 5-5:30 ET. So no worries if it would be past your bedtime. 

Just let me know what would work best for you.

**Pull request merged**